tests/extras/datasets/hdf5/test_h5py_dataset.py

```diff
@@ -0,0 +1,259 @@
+from pathlib import Path, PurePosixPath
+from typing import Union
+
+import h5py
+import numpy as np
+import pytest
+from fsspec.implementations.http import HTTPFileSystem
+from fsspec.implementations.local import LocalFileSystem
+from gcsfs import GCSFileSystem
+from s3fs.core import S3FileSystem
+
+from kedro.extras.datasets.hdf5.h5py_dataset import H5pyDataSet
+from kedro.io import DataSetError
+from kedro.io.core import PROTOCOL_DELIMITER, Version
+
+
+@pytest.fixture
+def filepath_hdf(tmp_path):
+    return (tmp_path / "test.h5").as_posix()
+
+
+@pytest.fixture
+def hdf_data_set(filepath_hdf, load_args, save_args, mocker, fs_args):
```

Comment:
Perhaps name this `h5py_data_set` to differentiate more from the existing hdf dataset.